common/src/main/java/dev/ftb/mods/ftbranks/FTBRanksCommands.java

```diff
@@ -171,17 +210,86 @@ private static int listRanksOf(CommandSourceStack source, ServerPlayer player) {
 		return 1;
 	}
 
-	private static int listPlayersWith(CommandSourceStack source, String name) {
-		Rank r = FTBRanksAPIImpl.manager.getRank(normalizeRankName(name)).orElseThrow(NullPointerException::new);
-
-		source.sendSuccess(new TextComponent("Players with " + name + " added to them:"), false);
+	private static int listPlayersWith(CommandSourceStack source, Rank rank) {
+		source.sendSuccess(new TranslatableComponent("ftbranks.list_players_with", rank.getName()), false);
 
 		for (ServerPlayer player : source.getServer().getPlayerList().getPlayers()) {
-			if (r.isActive(player)) {
+			if (rank.isActive(player)) {
 				source.sendSuccess(new TextComponent("- ").withStyle(ChatFormatting.YELLOW).append(player.getDisplayName()), false);
 			}
 		}
 
 		return 1;
 	}
+
+	private static int setNode(CommandSourceStack source, Rank rank, String node, String value) throws CommandSyntaxException {
+		try {
+			rank.setPermission(node, strToPermissionValue(value));
+			if (value != null) {
+				source.sendSuccess(new TranslatableComponent("ftbranks.node_added", node, rank.getPermission(node), rank), false);
+			} else {
+				source.sendSuccess(new TranslatableComponent("ftbranks.node_removed", node, rank), false);
+			}
+		} catch (IllegalArgumentException e) {
+			throw new SimpleCommandExceptionType(new TextComponent(e.getMessage())).create();
+		}
+
+		return 1;
+	}
+
+	private static int setCondition(CommandSourceStack source, Rank rank, String value) throws CommandSyntaxException {
+		try {
+			RankCondition condition;
+			if (value.equals("default") || value.equals("\"\"")) {
+				condition = new DefaultCondition(rank);
+			} else if (value.startsWith("{") || value.contains(" ")) {
+				condition = FTBRanksAPI.INSTANCE.getManager().createCondition(rank, SNBT.readLines(Collections.singletonList(value)));
+			} else {
+				condition = FTBRanksAPI.INSTANCE.getManager().createCondition(rank, StringTag.valueOf(value));
+			}
+			rank.setCondition(condition);
+			source.sendSuccess(new TranslatableComponent("ftbranks.node_added", "condition", value, rank), false);
+		} catch (Exception e) {
+			throw new SimpleCommandExceptionType(new TextComponent(e.getMessage())).create();
+		}
+
+		return 1;
+	}
+
+	private static int showRank(CommandSourceStack source, Rank rank) {
+		source.sendSuccess(new TextComponent(Strings.repeat('=', 50)).withStyle(ChatFormatting.GREEN), false);
```

Comment:
I originally missed this in the 1.19 PR, but this should probably not be using jopt.internal >.>

Reply:
Oops, yeah.  Meant to use `com.google.common.base.Strings` there

Reply:
I think starting in Java... 9? 11? You can also just use String.repeat from Java's stdlib? :P